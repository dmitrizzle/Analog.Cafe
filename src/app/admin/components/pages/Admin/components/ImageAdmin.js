import { ButtonStrip } from "@roast-cms/react-button-beans"
import React from "react"

import { GridButtonImage } from "../../../../../user/components/controls/Grid"
import {
  GridContainer,
  GridRow
} from "../../../../../user/components/controls/Grid/styles"
import Button from "../../../../../core/components/controls/Button/components/Button"
import ButtonStripItem from "../../../../../core/components/controls/Button/components/ButtonStripItem"

export default props => {
  return [
    <div style={{ padding: "0 1.5em" }} key="ImageAdmin_header">
      <h3>Images.</h3>
      <ButtonStrip style={{ margin: ".5em auto" }}>
        <div>
          <ButtonStripItem
            left
            inverse={
              !props.stateImageList.options.featured &&
              !props.stateImageList.options.fullConsent
            }
            onClick={view => props.imagesSwitchView("all")}
          >
            All
          </ButtonStripItem>
          <ButtonStripItem
            inverse={
              !props.stateImageList.options.featured &&
              props.stateImageList.options.fullConsent
            }
            onClick={view => props.imagesSwitchView("open")}
          >
            Open
          </ButtonStripItem>
          <ButtonStripItem
            right
            inverse={
              props.stateImageList.options.featured &&
              props.stateImageList.options.fullConsent
            }
            onClick={view => props.imagesSwitchView("feature")}
          >
            Featured
          </ButtonStripItem>
        </div>
      </ButtonStrip>
    </div>,
    <GridContainer key="ImageAdmin_grid">
      {props.rowIndex.map(rowNumber => (
        <GridRow key={rowNumber}>
          {props.imagelib.items
            .slice(
              rowNumber * props.gridRows,
              rowNumber * props.gridRows + props.gridRows
            )
            .map(item => (
              <GridButtonImage
                label={item.featured && "Featured"}
                span={4}
                noShim
                key={item.id}
                src={item.id}
                status={props.imagelib.status}
                author={props.imagelib.items[1] ? item.author : null}
                add={src =>
                  props.setCard({
                    info: {
                      title: src,
                      image: src,
                      buttons: [
                        {
                          to: `https://cloudinary.com/console/media_library/asset/image/upload/${src}`,
                          text: "View on Cloudinary"
                        },
                        {
                          to: "#image-delete",
                          onClick: event => props.imageDelete(src, event),
                          text: "Delete from Database"
                        },
                        props.stateImageList.options.fullConsent &&
                        !item.featured
                          ? {
                              to: "#image-feature",
                              onClick: event => props.imageFeature(src, event),
                              text: "Add to Featured List"
                            }
                          : null,
                        props.stateImageList.options.fullConsent &&
                        item.featured
                          ? {
                              to: "#image-unfeature",
                              onClick: event =>
                                props.imageUnfeature(src, event),
                              text: "Remove from Featured List"
                            }
                          : null
                      ]
                    }
                  })
                }
              />
            ))}
        </GridRow>
      ))}
    </GridContainer>,
    <div key="ImageAdmin_loadmore">
      {props.imagelib.page &&
        props.imagelib.page.total >= props.stateImageList.page && (
          <Button onClick={props.imagesLoadMore}>
            Load More ({props.imagelib.page["items-total"]})
          </Button>
        )}
    </div>
  ]
}
