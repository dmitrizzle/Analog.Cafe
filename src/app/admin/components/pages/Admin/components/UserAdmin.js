import React from "react"

import { GridButtonImage } from "../../../../../user/components/controls/Grid"
import {
  GridContainer,
  GridRow
} from "../../../../../user/components/controls/Grid/styles"

export default props => {
  return [
    <div style={{ padding: "0 1.5em" }} key="UserAdmin_header">
      <h3>Users.</h3>
    </div>,
    <GridContainer key="UserAdmin_grid">
      {props.rowIndex.map(rowNumber => (
        <GridRow key={rowNumber}>
          {props.admin.accountList.items
            .slice(
              rowNumber * props.gridRows,
              rowNumber * props.gridRows + props.gridRows
            )
            .map(item => (
              <GridButtonImage
                label={item.role}
                span={4}
                noShim
                key={item.id}
                src={item.image}
                status={props.admin.accountList.status}
                author={{
                  name: item.title,
                  id: item.id,
                  email: item.email
                }}
                add={(src, author) =>
                  props.setModal({
                    info: {
                      title: author.title || author.id,
                      text: `${author.email}`,
                      buttons: [
                        {
                          to: `mailto:${author.email}`,
                          text: "Contact via Email"
                        },
                        {
                          to: "#user-suspend",
                          onClick: () => alert("suspend"),
                          text: "Suspend user"
                        }
                      ]
                    }
                  })
                }
              />
            ))}
        </GridRow>
      ))}
    </GridContainer>
  ]
}
