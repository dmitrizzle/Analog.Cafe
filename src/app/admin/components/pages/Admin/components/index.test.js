import { shallow } from "enzyme"
import React from "react"

import ImageAdmin from "./ImageAdmin"
import UserAdmin from "./UserAdmin"

it("Render UserAdmin without crashing", () => {
  shallow(
    <UserAdmin
      rowIndex={[0]}
      admin={{
        accountList: {
          status: "loading",
          items: [{ id: 0 }]
        }
      }}
    />
  )
})

it("Render ImageAdmin without crashing", () => {
  shallow(
    <ImageAdmin
      stateImageList={{ options: { featured: false, fullConsent: false } }}
      rowIndex={[0]}
      imagelib={{
        status: "loading",
        items: [{ id: 0 }]
      }}
    />
  )
})
