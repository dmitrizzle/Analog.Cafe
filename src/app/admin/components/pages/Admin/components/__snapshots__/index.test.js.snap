// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render ImageAdmin without crashing, matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    imagelib={
        Object {
            "items": Array [
              Object {
                "id": 0,
              },
            ],
            "status": "loading",
          }
    }
    rowIndex={
        Array [
            0,
          ]
    }
    stateImageList={
        Object {
            "options": Object {
              "featured": false,
              "fullConsent": false,
            },
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Array [
    <div
      style={
            Object {
                  "padding": "0 1.5em",
                }
      }
>
      <h3>
            Images.
      </h3>
      <styled.div
            style={
                  Object {
                        "margin": ".5em auto",
                        "whiteSpace": "nowrap",
                      }
            }
      >
            <div>
                  <Unknown
                        inverse={true}
                        left={true}
                        onClick={[Function]}
                  >
                        All
                  </Unknown>
                  <Unknown
                        inverse={false}
                        onClick={[Function]}
                  >
                        Open
                  </Unknown>
                  <Unknown
                        inverse={false}
                        onClick={[Function]}
                        right={true}
                  >
                        Featured
                  </Unknown>
            </div>
      </styled.div>
</div>,
    <styled.div>
      <styled.div>
            
      </styled.div>
</styled.div>,
    <div />,
  ],
  Symbol(enzyme.__nodes__): Array [
    Array [
      <div
        style={
                Object {
                        "padding": "0 1.5em",
                      }
        }
>
        <h3>
                Images.
        </h3>
        <styled.div
                style={
                        Object {
                                "margin": ".5em auto",
                                "whiteSpace": "nowrap",
                              }
                }
        >
                <div>
                        <Unknown
                                inverse={true}
                                left={true}
                                onClick={[Function]}
                        >
                                All
                        </Unknown>
                        <Unknown
                                inverse={false}
                                onClick={[Function]}
                        >
                                Open
                        </Unknown>
                        <Unknown
                                inverse={false}
                                onClick={[Function]}
                                right={true}
                        >
                                Featured
                        </Unknown>
                </div>
        </styled.div>
</div>,
      <styled.div>
        <styled.div>
                
        </styled.div>
</styled.div>,
      <div />,
    ],
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Render UserAdmin without crashing, matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    admin={
        Object {
            "accountList": Object {
              "items": Array [
                Object {
                  "id": 0,
                },
              ],
              "status": "loading",
            },
          }
    }
    rowIndex={
        Array [
            0,
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Array [
    <div
      style={
            Object {
                  "padding": "0 1.5em",
                }
      }
>
      <h3>
            Users.
      </h3>
</div>,
    <styled.div>
      <styled.div>
            
      </styled.div>
</styled.div>,
  ],
  Symbol(enzyme.__nodes__): Array [
    Array [
      <div
        style={
                Object {
                        "padding": "0 1.5em",
                      }
        }
>
        <h3>
                Users.
        </h3>
</div>,
      <styled.div>
        <styled.div>
                
        </styled.div>
</styled.div>,
    ],
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
